internal/plugin/connectors/http/generic/config.go

```diff
@@ -85,23 +87,45 @@ func newConfig(cfgYAML *ConfigYAML) (*config, error) {
 		cfg.CredentialPatterns[cred] = re
 	}
 
-	// Validate and save header template strings
-	for header, tmplStr := range cfgYAML.Headers {
-		tmpl := newHeaderTemplate(header)
+	cfg.Headers, errs = validateAndSaveTemplateString(cfgYAML.Headers, errs)
+	cfg.QueryParams, errs = validateAndSaveTemplateString(cfgYAML.QueryParams, errs)
+
+	if err := errs.Filter(); err != nil {
+		return nil, err
+	}
+
+	return cfg, nil
+}
+
+func validateAndSaveTemplateString(templates map[string]string,
+	errs validation.Errors) (map[string]*template.Template, validation.Errors) {
```

Comment:
This is an odd indentation style. Maybe something like this for readability would be better:
```golang
func foo(
    blah int,
    baz int,
)(blahblah int) {

    logicstuff()
}
```

Reply:
Totally agree, fixed it up 👍